src/RuleCommonITILObject.php

```diff
@@ -371,61 +371,105 @@ public function executeActions($output, $params, array $input = [])
                         break;
 
                     case 'regex_result':
-                        if ($action->fields["field"] == "_affect_itilcategory_by_code") {
-                            if (isset($this->regex_results[0])) {
-                                $regexvalue = RuleAction::getRegexResultById(
-                                    $action->fields["value"],
-                                    $this->regex_results[0]
-                                );
-                            } else {
-                                $regexvalue = $action->fields["value"];
-                            }
+                        // Get each regex values
+                        $regex_values = array_map(
+                            fn ($regex_result) => RuleAction::getRegexResultById(
+                                $action->fields["value"],
+                                $regex_result
+                            ),
+                            $this->regex_results
+                        );
 
-                            if (!is_null($regexvalue)) {
-                                $target_itilcategory = ITILCategory::getITILCategoryIDByCode($regexvalue);
+                        // Get field
+                        $field = $action->fields["field"];
+
+                        // Handle each fields
+                        if ($field == "_affect_itilcategory_by_code") {
+                            $regex_value = $regex_values[0];
```

Comment:
In previous code, and in other pieces of code in rule engine, `if (isset($this->regex_results[0]))` check is done. I do not really know well this part, but I guess it may happen when regex does capture anything (i.e. there is no capture group, e.g. `/\d+/`, or capture group is optionnal, e.g. `/^asset-[a-z]+((?:-)\d+)?/`). Both are probably not correct in rule engine context, but are still correct regex.

So you should probably handle this.

Reply:
The previous behavior is using a fallback on `$action->fields["value"]`:
```php
if (isset($this->regex_results[0])) {
    $regexvalue = RuleAction::getRegexResultById(
        $action->fields["value"],
        $this->regex_results[0]
    );
} else {
    $regexvalue = $action->fields["value"];
}
```

This value will contain the raw regex and will (hopfully) not match any groups/appliance/type/category.
In this case, maybe we should just stop the execution ?
Meaning something like that:
```php
if (empty($this->regex_results)) {
    break;
}
```

Reply:
Sometimes, people are the most unlikely capabilities of a tool. We should try to ensure that we keep existing behaviour.

Reply:
I agree, but we should probably start by adding a warning or a deprecation, depending on the context, and then remove the code in the next major version.